src/main/java/dev/blaauwendraad/masker/json/KeyContainsMasker.java

```diff
@@ -39,9 +42,31 @@ final class KeyContainsMasker implements JsonMasker {
      */
     @Override
     public byte[] mask(byte[] input) {
-        try {
-            MaskingState maskingState = new MaskingState(input, !maskingConfig.getTargetJsonPaths().isEmpty());
+        MaskingState maskingState = new MaskingState(input, !maskingConfig.getTargetJsonPaths().isEmpty());
+        return mask(maskingState);
+    }
+
+    /**
+     * Runs masker in a streaming mode.
+     * The masker buffers data from provided input stream in chunks of size 8192 bytes and processes each chunk
+     * sequentially. The output is written into provided output stream after processing each chunk.
+     *
+     * @param inputStream the JSON input stream
+     * @param outputStream masked JSON output stream
+     */
+    @Override
+    public void mask(InputStream inputStream, OutputStream outputStream) {
+        MaskingState maskingState = new MaskingState(inputStream, outputStream, !maskingConfig.getTargetJsonPaths().isEmpty());
+        mask(maskingState);
+
+        // in case we are given a json that has trailing characters after the outermost object/array is closed
```

Comment:
I don't fully get why it is only if the character is repeated more than 8129 times why not if there is 8000 bytes of message plus 200 bytes of trailing white spaces. Then this also needs to happen, right?

Reply:
It is not bound to the buffer size. Internally, `readNextBuffer` first flushes the old buffer into the output stream and then reads the next one. So the while loop below will basically read the remaining of the input stream and flush them into the output stream in chunks of `bufferSize` length.
So even if there is only 1 whitespace character left in the input stream, the character gets transfered to the output stream.  
Also I agree that `readNextBuffer` is not a good name for a method because it does not imply that the method also flushes the old buffer, I will rename it.

Reply:
Done

Reply:
I have been thinking about this "while" loop for a while and decided not to transfer anything to the output stream after the outermost JSON object/array closure. I think that makes more sense.